Common/cpp/Tools/SingleInstanceChecker.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+namespace reanimated {
+
+using namespace std;
+
+// This is a class that counts how many instances of a different class there
+// are. It is meant only to be used with classes that should only have one
+// instance.
+
+template <class T>
+class SingleInstanceCheker {
+ public:
+  SingleInstanceCheker();
+  ~SingleInstanceCheker();
+
+ private:
+  void assertWithMessage(bool condition, string message) {
+    if (!condition) {
+      cerr << message << endl;
+      assert(condition);
+    }
+  }
+
+  // A static field will exist separatley for every class template.
+  inline static int instanceCount;
+};
+
+template <class T>
+SingleInstanceCheker<T>::SingleInstanceCheker() {
+  // This gives us a slightly magled class name, but it is still readable
```

Comment:
```suggestion
  // This gives us a slightly mangled class name, but it is still readable
```

Reply:
Also, feel free to check implementation of `ShareableValue::demangleExceptionName`.